api/src/main/java/org/openmrs/api/db/hibernate/HibernateLocationDAO.java

```diff
@@ -345,7 +345,7 @@ public LocationAttributeType getLocationAttributeTypeByName(String name) {
 		    Projections.rowCount()).add(Restrictions.eqProperty("alias.locationId", "outer.locationId"));
 		
 		return sessionFactory.getCurrentSession().createCriteria(Location.class, "outer").add(
-		    Restrictions.eq("retired", false)).add(Subqueries.eq((long) tags.size(), numberOfMatchingTags)).list();
+		    Restrictions.eq("retired", false)).add(Subqueries.eq(Long.valueOf(tags.size()), numberOfMatchingTags)).list();
```

Comment:
Looks good but could you edit your commit message to more intuitive wording of what you really did not just what you were supposed to do. Starting with something like "GCI-246 : Replaced.."

Reply:
Will do.